compiler/tests-common-new/tests/org/jetbrains/kotlin/test/runners/AbstractForeignAnnotationsTest.kt

```diff
@@ -35,10 +32,36 @@ import org.jetbrains.kotlin.test.services.jvm.PsiClassFilesReadingForCompiledJav
 import org.jetbrains.kotlin.test.services.sourceProviders.AdditionalDiagnosticsSourceFilesProvider
 import org.jetbrains.kotlin.test.services.sourceProviders.CoroutineHelpersSourceFilesProvider
 
-abstract class AbstractForeignAnnotationsTestBase : AbstractKotlinCompilerTest() {
+private val configureClassicFrontend: TestConfigurationBuilder.() -> Unit = {
+    globalDefaults {
+        frontend = FrontendKinds.ClassicFrontend
+    }
+
+    useMetaInfoProcessors(::OldNewInferenceMetaInfoProcessor)
+    classicFrontendStep()
+    classicFrontendHandlersStep {
+        useHandlers(
+            ::DeclarationsDumpHandler,
+            ::ClassicDiagnosticsHandler,
+        )
+    }
+
+    forTestsMatching("compiler/testData/diagnostics/foreignAnnotationsTests/java8Tests/jspecify/*") {
+        configureClassicFrontendHandlersStep {
+            useHandlers(::JspecifyDiagnosticComplianceHandler)
+        }
+        useSourcePreprocessor(::JspecifyMarksCleanupPreprocessor)
+    }
+
+    useAfterAnalysisCheckers(::FirTestDataConsistencyHandler)
+}
+
+abstract class AbstractForeignAnnotationsTestBase(
+    private val configureFrontend: TestConfigurationBuilder.() -> Unit = configureClassicFrontend
```

Comment:
Please make this `val` an `abstract fun`. Lambdas which are passed to constructors are counterintuitive

Reply:
Done in #4538

Reply:
Will do this by separate commit